api/src/main/java/org/apache/iceberg/metrics/IntCounter.java

```diff
@@ -45,8 +44,8 @@ public void increment(Integer amount) {
   }
 
   @Override
-  public Optional<Integer> count() {
-    return Optional.of(counter.get());
+  public Integer count() {
```

Comment:
Since this is in API, we can't remove it without deprecation. For now, can you add it back and just always return a non-empty optional? Then we can add a new method that always returns the value, like `value()`.

Reply:
Why did the `revapi` check pass with this?  Shouldn't it catch this type of change?

Reply:
Yes, it should!

Reply:
I looked a bit why Revapi didn't fail here. Turns out https://github.com/apache/iceberg/blob/master/.palantir/revapi.yml#L2 still compares the API to the 0.13.0 version, which didn't have the `Counter` API.

@rdblue can you please create `release-base-0.14.0`? 
I created https://github.com/apache/iceberg/pull/5336 that tells RevAPI to check API compatibility against `release-base-0.14.0`. Additionally, `release-base-0.14.0` is required so that nightly snapshot publishing uses `0.15.0-SNAPSHOT` (currently it still does `0.14.0-SNAPSHOT`, because of https://github.com/apache/iceberg/blob/8300b219ec8962bd22af69adf67a4ab72dec1f48/build.gradle#L682).